imports/ui/ranking/Ranking.js

```diff
@@ -1,21 +1,125 @@
 import GameTypeUtils from '../../api/utils/GameTypeUtils';
+import Constants from '../../api/Constants';
 
 import '../statistics/PlayerModal';
 import './HongKongRanking';
 import './JapaneseRanking';
 import './Ranking.html';
 
+Template.Ranking.onCreated(
+    /**
+     * Setup "constructor" for Ranking page, ensure that default stats are shown
+     */
+    function() {
+        Session.set("sortStatistic", "elo");
+        Session.set("additionalStatistic", "gamesPlayed");
+    }
+);
+
 Template.Ranking.helpers({
     getName(format) {
         return GameTypeUtils.formatName(format);
     },
 
-    getPlayers(format, sort) {
-        return GameTypeUtils.getPlayers(format, sort).map((p, i) => ({
+    /**
+     * Return the list of valid players, ranked by ELO and sorted by orderSort
+     * @param[in] format The Constants.GAME_TYPE game format
+     * @param[in] rankingSort The order to rank players by (currently always ELO)
+     * @param[in] orderSort The order to sort players by
+     * @returns A list of players with statistics
+     */
+    getOrderedPlayers(format, rankingSort, orderSort) {
+        return GameTypeUtils.getPlayers(format, rankingSort).map((p, i) => ({
             ...p,
             elo: p.elo.toFixed(3),
             rank: i + 1
-        }));
+        })).sort(orderSort.hash.orderSort);
+    },
+
+    /**
+     * List generator for ranking page statistics options
+     * @param[in] format The Constants.GAME_TYPE format
+     * @param[in] exclusions List of statistics to exclude by value (Ignore format!)
+     * @returns an object of formatted statistics keyed with statistics
+     */
+    getRankingStatistics(format, exclusions) {
+        let statisticsList = [
+            {value: "elo", format: "ELO"},
+            {value: "gamesPlayed", format: "Games"},
+            {value: "handWinRate", format: "Hand Win %"},
+            {value: "dealinRate", format: "Deal-in %"},
+            {value: "averageHandSize", format: "Avg Hand Size"},
+            {value: "averagePosition", format: "Avg Position"},
+            {value: "flyRate", format: "Bankrupt %"},
+            {value: "chomboTotal", format: "Chombos"}
+        ];
+
+        if (format === Constants.GAME_TYPE.JAPANESE) {
+            const japaneseStatisticsList = [
+                {value: "averageHandDora", format: "Avg Hand Dora"},
+                {value: "riichiRate", format: "Riichi %"},
+                {value: "riichiWinRate", format: "Riichi Win %"}
+            ];
+            statisticsList = statisticsList.concat(japaneseStatisticsList);
+        }
+
+        for (let stat of exclusions) {
```

Comment:
I need time to process this comment.  You're probably right though.  I thought Javascript was all about that list comprehension though?  Are for loops actually discouraged?

Reply:
Some people like `for...each`, but generally the React community tend to favor the functional approach with `map` etc. and I work with React everyday so I don't like for loops.

List comprehension might be a thing more recently, but generally the JIT compiler is so good now that performance is gonna be roughly the same no matter what you use

Reply:
I'm going to change this up a bit to maintain some modularity.  Since we might end up with different statistics for Cantonese style or any other style we might add in the future, I want to ensure we can operate on all formats.